main.js

```diff
@@ -232,15 +233,17 @@ define(function (require, exports, module) {
   function handleEditorChange ($event, newEditor, oldEditor) {
     
     if (oldEditor) {
-      oldEditor._codeMirror.off('change', sendLocalChange)
-      oldEditor._codeMirror.off('beforeSelectionChange', sendLocalSelect)
+      oldEditor._codeMirror.off('change', onCodeChange)
+      oldEditor._codeMirror.off('changes', onCodeChanges)
+      oldEditor._codeMirror.off('beforeSelectionChange', onBeforeSelectionChange)
     }
 
     if (newEditor) {
       currentEditor = newEditor
       documentRelativePath = FileUtils.getRelativeFilename(projectBasePath, newEditor.document.file.fullPath)
-      newEditor._codeMirror.on('change', sendLocalChange)
-      newEditor._codeMirror.on('beforeSelectionChange', sendLocalSelect)
+      newEditor._codeMirror.on('change', onCodeChange)
+      newEditor._codeMirror.on('changes', onCodeChanges)
```

Comment:
If we are going to start listening to operation-batched changes, we should only use the 'changes' event.